src/System.Console/src/System/ConsolePal.Unix.cs

```diff
@@ -251,25 +254,37 @@ private struct TerminalColorInfo
             /// <summary>The cached instance.</summary>
             public static TerminalColorInfo Instance { get { return _instance.Value; } }
 
-            /// <summary>Lazy initialization of the terminal color information.</summary>
-            private static Lazy<TerminalColorInfo> _instance = new Lazy<TerminalColorInfo>(() =>
+            internal TerminalColorInfo(TermInfo.Database db)
             {
-                TermInfo.Database db = TermInfo.Database.Instance;
-                TerminalColorInfo tci = new TerminalColorInfo();
                 if (db != null)
                 {
-                    tci.ForegroundFormat = db.GetString(TermInfo.Database.SetAnsiForegroundIndex);
-                    tci.BackgroundFormat = db.GetString(TermInfo.Database.SetAnsiBackgroundIndex);
-                    tci.ResetFormat =
+                    ForegroundFormat = db.GetString(TermInfo.Database.SetAnsiForegroundIndex);
+                    BackgroundFormat = db.GetString(TermInfo.Database.SetAnsiBackgroundIndex);
+                    ResetFormat =
                         db.GetString(TermInfo.Database.OrigPairsIndex) ??
                         db.GetString(TermInfo.Database.OrigColorsIndex);
 
                     int maxColors = db.GetNumber(TermInfo.Database.MaxColorsIndex);
-                    tci.MaxColors = // normalize to either the full range of all ANSI colors, just the dark ones, or none
+                    MaxColors = // normalize to either the full range of all ANSI colors, just the dark ones, or none
                         maxColors >= 16 ? 16 :
                         maxColors >= 8 ? 8 :
                         0;
                 }
+                else
+                {
+                    Debug.Assert(false);
+                    ForegroundFormat = default(string);
+                    BackgroundFormat = default(string);
+                    ResetFormat = default(string);
+                    MaxColors = default(int);
+                }
+            }
+
+            /// <summary>Lazy initialization of the terminal color information.</summary>
+            private static Lazy<TerminalColorInfo> _instance = new Lazy<TerminalColorInfo>(() =>
+            {
+                TermInfo.Database db = TermInfo.Database.Instance;
+                TerminalColorInfo tci = new TerminalColorInfo(db);
```

Comment:
Assuming you change the ctor to assert if db is null, then you'd change this to:
```C#
TermInfo.Database db = TermInfo.Database.Instance;
return db != null ?
    new TerminalColorInfo(db) :
    default(TerminalColorInfo);
```